pkg/cmd/roachtest/jepsen.go

```diff
@@ -210,16 +217,32 @@ cd /mnt/data1/jepsen/cockroachdb && set -eo pipefail && \
 		//
 		// Try to get any running jvm to log its stack traces for
 		// extra debugging help.
-		run(c, ctx, controller, "pkill -QUIT java")
+		run(c, ctx, controller, nil /* stdout */, nil /* stderr */, "pkill -QUIT java")
 		time.Sleep(10 * time.Second)
-		run(c, ctx, controller, "pkill java")
+		run(c, ctx, controller, nil /* stdout */, nil /* stderr */, "pkill java")
 		logf("timed out")
 		testErr = fmt.Errorf("timed out")
 	}
 
 	if testErr != nil {
 		logf("grabbing artifacts from controller. Tail of controller log:")
-		run(c, ctx, controller, "tail -n 100 /mnt/data1/jepsen/cockroachdb/invoke.log")
+		// We'll capture the output and recognize some errors.
+		// TODO(andrei): Remove this unfortunate code once #30527 is fixed.
+		stdoutBuf, _ := circbuf.NewBuffer(1 << 20) // 1MB
+		run(c, ctx, controller, stdoutBuf, stdoutBuf,
+			"tail -n 100 /mnt/data1/jepsen/cockroachdb/invoke.log")
+		s := stdoutBuf.String()
+		ignoreErr := false
+		if strings.Contains(s, "java.lang.InterruptedException") ||
```

Comment:
I believe the logs always contain some InterruptedExceptions. I think what we want to look for is the message "Oh jeez, I'm sorry, Jepsen broke. Here's why:" followed by a line containing InterruptedException or BrokenBarrierException. 

Generally, Jepsen will produce one of three messages for the outcome of the test:

- "Everything looks good! ヽ(‘ー`)ノ"
- "Analysis invalid! (ﾉಥ益ಥ）ﾉ ┻━┻")))
- "Oh jeez, I'm sorry, Jepsen broke. Here's why:"

"Analysis invalid" is a real consistency failure; "Oh jeez" is any other error (most often in setup or shutdown, which is usually benign, although the risk with ignoring these is that we may be fruitlessly running the tests without getting a clear signal either way). 

These lines are not guaranteed to appear within the last 100 lines. I think we need to cat the whole invoke.log instead of just the last 100 lines.

Reply:
Something we can do to highlight the benign errors is to add a `test.Skip()` method that sets `test.spec.Skip` so that the benign failures are marked as skipped instead of passed.